experimental/token/cursor.go

```diff
@@ -27,44 +26,40 @@ import (
 type Cursor struct {
 	withContext
 
-	// This is used if this is a cursor over natural tokens.
-	// start is inclusive, end is exclusive. start == end means the stream
-	// is empty.
-	start, end ID
 	// This is used if this is a cursor over the children of a synthetic token.
 	// If stream is nil, we know we're in the natural case.
 	stream []ID
-	idx    int
+	// The idx is the current token ID if this is a cursor over natural tokens,
+	// or the index into stream if this is a cursor over synthetic tokens.
+	idx int
```

Comment:
I think the code below might be simplified if this is specified to be an index into `stream.nats` (i.e., a token ID minus one).

Reply:
Done

Reply:
Typo: `or` -> `of`. Maybe just write `Context().Stream().nats`?